pkg/networkmanager/firewall.jsx

```diff
@@ -408,60 +389,63 @@ class AddServicesModal extends React.Component {
     }
 
     validate(event) {
-        const ports = event.target.value.split(',');
         let error = "";
         let targets = ['tcp', 'custom_tcp_ports', 'tcp_error', 'custom_tcp_value'];
         if (event.target.id === "udp-ports")
             targets = ['udp', 'custom_udp_ports', 'udp_error', 'custom_udp_value'];
-
         const new_ports = [];
         const self = this;
-        ports.forEach(function(port) {
-            port = port.trim();
-            if (!port)
-                return;
-            let ports;
-            if (port.indexOf("-") > -1) {
-                ports = port.split("-");
-                if (ports.length != 2) {
-                    error = _("Invalid range");
+        const event_value = event.target.value;
+        const event_id = event.target.id;
+
+        this.setState(oldState => {
```

Comment:
Just OOI, what is the benefit of having a functional setState() here? (I'm not saying it's wrong/bad, I'm just not aware of  its advantage)

Reply:
Well the new state is based on the old state. And I thought to avoid Object.assign side-effects you have to use the functional setState.